main_test.go

```diff
@@ -770,11 +811,88 @@ maps.Values from Python dictionary: []float64{3, 5}
 
 func TestBindStrings(t *testing.T) {
 	t.Parallel()
+	path := "_examples/gostrings"
 	testPkg(t, pkg{
-		path: "_examples/gostrings",
-		lang: []string{"py2", "py2-cffi", "py3-cffi", "pypy2-cffi", "pypy3-cffi"},
+		path: path,
+		lang: features[path],
 		want: []byte(`S1 = S1
 GetString() = MyString
 `),
 	})
 }
+
+// Generate / verify SUPPORT_MATRIX.md from features map.
+func TestCheckSupportMatrix(t *testing.T) {
+	var supportMatrixMd bytes.Buffer
+	supportMatrixMd.WriteString("# Support matrix\n")
+	supportMatrixMd.WriteString(`
+NOTE: File auto-generated by TestCheckSupportMatrix in main_test.go. Please
+don't modify manually.
+
+`)
+
+	// Generate
+	// - sorted list of features
+	// - sorted list of backends
+	// - a map of feature to available backends
+	var featuresSorted []string
+	var allBackendsSorted []string
+	featureToBackendMap := make(map[string]map[string]bool)
+	allBackends := make(map[string]bool)
+	for feature, backends := range features {
+		featuresSorted = append(featuresSorted, feature)
+		featureToBackendMap[feature] = make(map[string]bool)
+		for _, backend := range backends {
+			featureToBackendMap[feature][backend] = true
+			allBackends[backend] = true
+		}
+	}
+	for backend, _ := range allBackends {
+		allBackendsSorted = append(allBackendsSorted, backend)
+	}
+	sort.Strings(featuresSorted)
+	sort.Strings(allBackendsSorted)
+
+	// Write the table header and the line separating header and rows.
+	supportMatrixMd.WriteString("Feature |")
+	supportMatrixMd.WriteString(strings.Join(allBackendsSorted, " | "))
+	supportMatrixMd.WriteString("\n")
+	var tableDelimiters []string
+	for i := 0; i <= len(allBackendsSorted); i++ {
+		tableDelimiters = append(tableDelimiters, "---")
+	}
+	supportMatrixMd.WriteString(strings.Join(tableDelimiters, " | "))
+	supportMatrixMd.WriteString("\n")
+
+	// Write the actual rows of the support matrix.
+	for _, feature := range featuresSorted {
+		var cells []string
+		cells = append(cells, feature)
+		for _, backend := range allBackendsSorted {
+			if featureToBackendMap[feature][backend] {
+				cells = append(cells, "yes")
+			} else {
+				cells = append(cells, "no")
+			}
+		}
+		supportMatrixMd.WriteString(strings.Join(cells, " | "))
+		supportMatrixMd.WriteString("\n")
+	}
+
+	if os.Getenv("GOPY_GENERATE_SUPPORT_MATRIX") == "1" {
+		ioutil.WriteFile("SUPPORT_MATRIX.md", supportMatrixMd.Bytes(), 0644)
```

Comment:
please check the returned error.
also, add an early return so we don't have to have the `else` branch below.

Reply:
Done.